TheJumpCloud/support
# SA-2864: Update CrowdStrike Install Command Scripts to Utilize CrowdStrike API

## Issues
* [SA-2864](https://jumpcloud.atlassian.net/browse/SA-2864) - Update CrowdStrike Install Command Scripts to Utilize CrowdStrike API

## What does this solve?
Previously, we were instructing customers to download the sensor file and host it publicly in a different location so that the local workstations can access the URL. 

With the JumpCloud x CrowdStrike partnership, customers should be able to utilize CrowdStrike's API to find the latest version of the installer and download it directly from their resources rather than having to host it in an outside location.

## Is there anything particularly tricky?
When testing on Mac devices, ensure that the **CrowdStrike Falcon MDM Policy is correctly applied to devices BEFORE running the command**, otherwise it will not work

## How should this be tested?
1. Navigate to your CrowdStrike console
2. Create a new CrowdStrike API Client with Sensor Download - Read Scope
    * Click left hamburger menu
    * Select **Support and Resources**
    * Under **Resources and Tools**, select **API clients and keys**
    * Select **Add new API Client**
        * Fill out Client Name and Description
        * Scroll down to **Sensor Download** and check the **Read** scope box
        * Select **Add**
3. Save the **Client ID** and **Secret** and **Base URL** Information for future use
4. Create a new Command in the JumpCloud console for each of the installations (Mac and Windows)
    * Make the following adjustments either directly in the Command or in the command settings
        * **MAC** - Run As Root
        * **WINDOWS** - Check Windows PowerShell
        * Replace **CSBaseAddress**, **CSClientID** and **CSClientSecret** with information saved in **Step 3**.
        * Change Timeout Period to something in the neighborhood of 300-600 seconds
5. Assign the command to devices
6. Execute command

## Screenshots
![image](https://user-images.githubusercontent.com/89030113/195150002-662bcc85-6696-415c-ac75-b04628b1274d.png)
![image](https://user-images.githubusercontent.com/89030113/195150124-6f732587-bdf6-4464-86e7-39605db75dfd.png)
